### ISSUE TITLE
[azurerm_linux_function_app] elastic_instance_minimum attribute not setting correctly

### ISSUE THREAD

**Issue by mkprizzle** [hashicorp/terraform-provider-azurerm#16995]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.1

### AzureRM Provider Version

3.7.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_function_app possibly azurerm_windows_function_app but untested

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "functionapprg"
  location = "East US 2"
}

resource "azurerm_storage_account" "example" {
  name                            = "storage_account_name"
  resource_group_name             = azurerm_resource_group.example.name
  location                        = azurerm_resource_group.example.location
  account_tier                    = "Standard"
  account_replication_type        = "LRS"
  allow_nested_items_to_be_public = false
}

resource "azurerm_service_plan" "example" {
  name                = "amazingfunctionplan"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  os_type             = "Linux"
  sku_name            = "EP2"
  # Issue persists also when true
  per_site_scaling_enabled = false
  # Separate issue, zone_balancing_enabled sets minimum to 3, subsequent applies overwrite
  zone_balancing_enabled = false
  maximum_elastic_worker_count = 10
}

resource "azurerm_linux_function_app" "example" {
  name                = "function_name"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  service_plan_id            = azurerm_service_plan.example.id
  storage_account_name       = azurerm_storage_account.example.name
  storage_account_access_key = azurerm_storage_account.example.primary_access_key

  site_config {
    elastic_instance_minimum = 5
    pre_warmed_instance_count = 1
    application_stack {
      python_version = "3.9"
      use_dotnet_isolated_runtime = null
    }
  }

  identity {
    type = "UserAssigned"
    identity_ids = [azurerm_user_assigned_identity.identity.id]
  }
}

resource "azurerm_user_assigned_identity" "identity" {
  name = "funcidentity"
  resource_group_name = azurerm_resource_group.example.name
  location = azurerm_resource_group.example.location
}
```

### Debug Output/Panic Output

```shell
No panic
```

### Expected Behaviour

Elastic minimum/pre warmed instances set correctly, quiet subsequent applies

### Actual Behaviour

TF apply does not set the setting, subsequent applies continue to try and set this.

Apply
![image](https://user-images.githubusercontent.com/37881072/170778581-49669f1a-6f04-4ead-8079-a7cc60bed367.png)
Outcome in portal
![image](https://user-images.githubusercontent.com/37881072/170778709-4a45c849-4e44-438e-811c-0fe9920f3516.png)
Subsequent plan
![image](https://user-images.githubusercontent.com/37881072/170778781-988bc014-19c6-4e41-964f-682a2164bf0c.png)

### Steps to Reproduce

TF apply provided sample code, check elastic minimum

**Comment by xiaxyi** [comment]
Hey @mkprizzle , can you do me a favor which is to check if the property is functioning well after the fix was merged? Thanks!

**Comment by mkprizzle** [comment]
> Hey @mkprizzle , can you do me a favor which is to check if the property is functioning well after the fix was merged? Thanks!

Confirmed fixed (tested on 3.18.0)
